shiny/express/_page.py

```diff
@@ -0,0 +1,334 @@
+from __future__ import annotations
+
+from typing import Callable, Literal, Optional, cast
+
+from htmltools import Tag, TagAttrValue, TagChild, TagList
+
+from .. import ui
+from ..types import MISSING, MISSING_TYPE
+from ..ui._navs import NavMenu, NavPanel
+from ..ui._sidebar import Sidebar
+from ..ui.css import CssUnit
+from ._recall_context import RecallContextManager
+from ._run import get_top_level_recall_context_manager
+
+__all__ = (
+    "page_auto",
+    "page_auto_cm",
+    "set_page_title",
+    "set_page_lang",
+    "set_page_fillable",
+    "set_page_wider",
+    "use_page_fixed",
+    "use_page_fluid",
+    "use_page_fillable",
+    "use_page_sidebar",
+    "use_page_navbar",
+)
+
+
+def page_auto_cm() -> RecallContextManager[Tag]:
+    return RecallContextManager(
+        page_auto,
+        kwargs={
+            "_page_fn": None,
+            "_fillable": False,
+            "_wider": False,
+        },
+    )
+
+
+def page_auto(
+    *args: object,
+    _page_fn: Callable[[object], Tag] | None,
+    _fillable: bool,
+    _wider: bool,
+    **kwargs: object,
+) -> Tag:
+    # Presence of a top-level nav items and/or sidebar determines the page function
+    navs = [x for x in args if isinstance(x, (NavPanel, NavMenu))]
+    sidebars = [x for x in args if isinstance(x, ui.Sidebar)]
+
+    nNavs = len(navs)
+    nSidebars = len(sidebars)
+
+    if _page_fn is None:
+        if nNavs == 0:
+            if nSidebars == 0:
+                if _fillable:
+                    _page_fn = (
+                        ui.page_fillable
+                    )  # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
Not that we absolutely need this now, but I think we could support `wider=F` when `fillable=T` by including  a `{"class": "container"}`?